tasks/staking.ts

```diff
@@ -36,53 +38,63 @@ export function load() {
   task("staking:single:pause")
     .addPositionalParam("stakingRewardAddress", "Staking reward pool contract address")
     .setAction(async ({ stakingRewardAddress }, hre) => {
-      const deployer = await getDeployer(hre);
-      const stakingRewardContract = (await hre.ethers.getContractAt("StakingRewards", stakingRewardAddress)) as StakingRewards;
-      if (!(await stakingRewardContract.connect(deployer).paused())) {
-        await (await stakingRewardContract.pause()).wait();
+      const recorder = await defaultRecorder(hre);
+
+      const stakingRewardContract = toRc((await hre.ethers.getContractAt("StakingRewards", stakingRewardAddress)) as StakingRewards, recorder);
+      if (!(await stakingRewardContract.paused())) {
+        await stakingRewardContract.record.pause();
         console.log(`StakingPool ${stakingRewardAddress} is now paused!`);
       } else {
         console.log(`StakingPool ${stakingRewardAddress} is already paused`);
       }
+      await recorder.execute();
     });
 
   task("staking:single:unpause")
     .addPositionalParam("stakingRewardAddress", "Staking reward pool contract address")
     .setAction(async ({ stakingRewardAddress }, hre) => {
-      const deployer = await getDeployer(hre);
-      const stakingRewardContract = (await hre.ethers.getContractAt("StakingRewards", stakingRewardAddress)) as StakingRewards;
-      if (await stakingRewardContract.connect(deployer).paused()) {
-        await (await stakingRewardContract.unpause()).wait();
-        console.log(`StakingPool ${stakingRewardAddress} is now unpaused!`);
+      const recorder = await defaultRecorder(hre);
+
+      const stakingRewardContract = toRc((await hre.ethers.getContractAt("StakingRewards", stakingRewardAddress)) as StakingRewards, recorder);
+      if (await stakingRewardContract.paused()) {
+        await stakingRewardContract.record.unpause();
       } else {
         console.log(`StakingPool ${stakingRewardAddress} is already unpaused`);
       }
+
+      await recorder.execute();
     });
 
   task("staking:all:pause").setAction(async (args, hre) => {
+    const recorder = await defaultRecorder(hre);
+
     const stakings = await getAllBDStableStakingRewards(hre);
     for (const staking of stakings) {
-      if (!(await staking.paused())) {
-        await staking.pause();
-        console.log("paused", staking.address);
+      const stikingRecordable = toRc(staking, recorder);
```

Comment:
stikingRecordable => stakingRecordable

Reply:
or maybe a better name will be `recordableStakingReward`
NIT - leaving it up to you to decide